modules/slugs/includes/class-qtranslate-slug.php

```diff
@@ -370,6 +372,9 @@ function filter_request( $query ) {
 
         // -> page
         elseif ( isset( $query['pagename'] ) || isset( $query['page_id'] ) ):
+            if ( isset( $query['name'] ) ) {
+                unset($query['name']);
+            }
```

Comment:
Why is the `name` removed?

Reply:
both posts and pages have as a minimum only `page` and `name` keys. No way to distinguish between them looking only at `$query` arg only.
Must look into `matched_query` for `pagename` key to identify a page, but if `$query` is returned both with `name` and `pagename` keys, still posts have the priority and you get a 404.